tests/suites/storage/charm_storage.sh

```diff
@@ -0,0 +1,146 @@
+		juju list-storage-pools -m "${model_name}" --format json | jq '.loop | .provider' | check "loop"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.rootfs | .provider' | check "rootfs"
+	else
+		juju list-storage-pools -m "${model_name}" --format json | jq '.rooty | .provider' | check "rootfs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.tempy | .provider' | check "tmpfs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.loopy | .provider' | check "loop"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.ebsy | .provider' | check "ebs"
+	fi
+	echo "Storage pool PASSED"
+
+	# Assess charm storage with the filesystem storage provider
+	echo "Assessing filesystem rootfs"
+	juju deploy ./testcharms/charms/dummy-storage-fs --series="jammy" --storage data=rootfs,1G
+	wait_for "dummy-storage-fs" ".applications"
+	# assert the storage kind name
+	juju list-storage --format json | jq '.storage[] | .kind' | check "filesystem"
+	# assert the storage label name
+	juju list-storage --format json | jq '.storage | keys | .[]' | check "data/0"
+	# assert the unit attachment name
+	juju list-storage --format json | jq '.storage[] | .attachments | .units | keys | .[]' | check "dummy-storage-fs/0"
+	# assert the attached unit state
+	juju list-storage --format json | jq '.storage[] | .attachments | .units[] | .life' | check "alive"
+	echo "Filesystem rootfs PASSED"
+	# remove the application
+	juju remove-application dummy-storage-fs
+	wait_for "{}" ".applications"
+
+	# Assess charm storage with the filesystem storage provider
+
+	echo "Assessing block loop disk 1"
+	juju deploy ./testcharms/charms/dummy-storage-lp --series="jammy" --storage disks=loop,1G
+	wait_for "dummy-storage-lp" ".applications"
+	# assert the storage kind name
+	juju list-storage --format json | jq '.storage | .["disks/1"] | .kind' | check "block"
+	# assert the storage label name
+	juju list-storage --format json | jq '.storage| keys[0]' | check "disks/1"
+	# assert the unit attachment name
+	juju list-storage --format json | jq '.storage |.["disks/1"]|.attachments |.units | keys | .[]' | check "dummy-storage-lp/0"
+	# assert the attached unit state
+	juju list-storage --format json | jq '.storage[] | .attachments |.units[] | .life' | check "alive"
+	echo "Block loop disk 1 PASSED"
+
+	echo "Assessing add storage block loop disk 2"
+	juju add-storage -m "${model_name}" dummy-storage-lp/0 disks=1
+	# assert the storage kind name
+	juju list-storage --format json | jq '.storage| .["disks/2"]| .kind' | check "block"
+	# assert the storage label name
+	juju list-storage --format json | jq '.storage | keys[1]' | check "disks/2"
+	# assert the unit attachment name
+	juju list-storage --format json | jq '.storage | .["disks/2"] | .attachments |.units | keys | .[]' | check "dummy-storage-lp/0"
+	# assert the attached unit state
+	juju list-storage --format json | jq '.storage | .["disks/2"] | .life' | check "alive"
```

Comment:
Similar sets of asserts are duplicated a fair few times. Could this be extracted into a function?

Reply:
But the type of storage provider is different.

Reply:
I am not an expert with the different storage providers, but it looks to me like the differences can probably be handled with parameters (different assert values)

I see there are some differences between each block of sets, ranging from small inconsistencies to different queries (which I suspect may be a mistake. Extracting a function would help with these